### ISSUE TITLE
Fixing travis

### ISSUE THREAD

**Issue by jchodera** [choderalab/yank#187]
Any idea what might be going on here with this travis failure?  I'm not even sure where the test is that is causing this failure.

https://travis-ci.org/choderalab/yank#L1357-L1371
```
======================================================================
ERROR: test suite for <module 'yank.commands' from '/home/travis/miniconda/envs/_test/lib/python2.7/site-packages/yank-0.9.0-py2.7-linux-x86_64.egg/yank/commands/__init__.py'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/_test/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/home/travis/miniconda/envs/_test/lib/python2.7/site-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/home/travis/miniconda/envs/_test/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/home/travis/miniconda/envs/_test/lib/python2.7/site-packages/nose/util.py", line 465, in try_run
    (name, obj))
TypeError: Attribute setup of <module 'yank.commands' from '/home/travis/miniconda/envs/_test/lib/python2.7/site-packages/yank-0.9.0-py2.7-linux-x86_64.egg/yank/commands/__init__.py'> is not a python function. Only functions or callables may be used as fixtures.
```

**Comment by jchodera** [comment]
OK, I'm very confused. Running `nosetests Yank --nocapture --verbosity=2 --with-doctest` on my laptop does show a bunch of different errors:
```
[LSKI1497:~/code/yank/yank.jchodera] choderaj% nosetests Yank --nocapture --verbosity=2 --with-doctest
Doctest: Yank.alchemy.AbsoluteAlchemicalFactory ... ok
Doctest: Yank.alchemy.AbsoluteAlchemicalFactory._is_restraint ... ok
Doctest: Yank.alchemy.AbsoluteAlchemicalFactory.createPerturbedSystem ... ok
Doctest: Yank.alchemy.AbsoluteAlchemicalFactory.createPerturbedSystems ... ok
Doctest: Yank.alchemy.AbsoluteAlchemicalFactory.perturbContext ... ok
Doctest: Yank.alchemy.AbsoluteAlchemicalFactory.perturbSystem ... ok
Doctest: Yank.alchemy.AlchemicalState.__init__ ... ok
Doctest: Yank.alchemy._is_periodic ... ok
ERROR
Failure: ImportError (No module named _mix_replicas) ... ERROR
Doctest: Yank.restraints.FlatBottomReceptorLigandRestraint ... FAIL
Doctest: Yank.restraints.HarmonicReceptorLigandRestraint ... FAIL
Failure: ImportError (No module named _mix_replicas) ... ERROR
Failure: ImportError (No module named _mix_replicas) ... ERROR
...
```

**Comment by kyleabeauchamp** [comment]
That looks like a possible install / import issue with the cython code.

**Comment by rmcgibbo** [comment]
Likely a confusion between the install directory in ../lib/site-packages/etc and imports from the current directory.

- Are you running the tests from inside the source directory?
- Are you using `setup.py develop` mode?

**Comment by kyleabeauchamp** [comment]
Also, for now I'm recommending that we test things *outside* the source tree, as that's the way end-users will be testing.

**Comment by jchodera** [comment]
> Also, for now I'm recommending that we test things outside the source tree, as that's the way end-users will be testing.

How do we test from outside the source tree?  I've been testing from the `yank` checkout, just outside the `Yank/` source directory?

**Comment by pgrinaway** [comment]
That's weird. Looking into this now.

**Comment by jchodera** [comment]
> Are you using setup.py develop mode?

Nope.

**Comment by kyleabeauchamp** [comment]
```
cd ~/
nosetests yank --nocapture --verbosity=2 --with-doctest -a '!slow'
```

**Comment by jchodera** [comment]
Trying this now. Presumably I can drop the `-a '!slow'`?

**Comment by jchodera** [comment]
Here's what I get instead of the mixer errors:
```
[LSKI1497:~] choderaj% nosetests yank --nocapture --verbosity=2 --with-doctest 
Doctest: yank.alchemy.AbsoluteAlchemicalFactory ... ok
Doctest: yank.alchemy.AbsoluteAlchemicalFactory._is_restraint ... ok
Doctest: yank.alchemy.AbsoluteAlchemicalFactory.createPerturbedSystem ... ok
Doctest: yank.alchemy.AbsoluteAlchemicalFactory.createPerturbedSystems ... ok
Doctest: yank.alchemy.AbsoluteAlchemicalFactory.perturbContext ... ok
Doctest: yank.alchemy.AbsoluteAlchemicalFactory.perturbSystem ... ok
Doctest: yank.alchemy.AlchemicalState.__init__ ... ok
Doctest: yank.alchemy._is_periodic ... ok
ERROR
Doctest: yank.repex.HamiltonianExchange ... /Users/choderaj/anaconda/lib/python2.7/site-packages/mdtraj/utils/validation.py:116: TypeCastPerformanceWarning: Casting Replica States dtype=int32 to <type 'numpy.int64'> 
  TypeCastPerformanceWarning)
/Users/choderaj/anaconda/lib/python2.7/site-packages/mdtraj/utils/validation.py:116: TypeCastPerformanceWarning: Casting Reduced Potentials dtype=float32 to <type 'numpy.float64'> 
  TypeCastPerformanceWarning)
ok
```

**Comment by rmcgibbo** [comment]
If you pass `-x` to nosetests, it will exit on the first error. Useful when debugging so you can see the whole traceback without waiting.

**Comment by kyleabeauchamp** [comment]
It's something about `nose` finding test fixtures inside our comands module, presumably from the line `import setup`

**Comment by kyleabeauchamp** [comment]
Commenting out `import setup` in `commands/__init__.py` seems to fix the issue.

**Comment by rmcgibbo** [comment]
I think you could fix this up by using explicit relative imports (e.g. `from . import setup`). With unqualified imports (just `import setup`, you're sort of playing with fire.)

**Comment by kyleabeauchamp** [comment]
I just tried switching to explicit relative imports in `commands/__init__.py`--that change does not address the issue.  AFAIK, I think there's also something fishy about nose's fixture detection.

**Comment by jchodera** [comment]
OK, so we'll fix by commenting out `import setup` in `commands/__init__.py` for now.  I'll open a new PR.

**Comment by kyleabeauchamp** [comment]
Wait, that is not a real fix though, because it disables the `yank setup` command.   Still need a complete workaround.

**Comment by kyleabeauchamp** [comment]
There are some related issues in `nose`, but I can't seem to find a workaround that actually works.

https://code.google.com/p/python-nose/issues/detail?id=326
https://code.google.com/p/python-nose/issues/detail?id=320
https://github.com/nose-devs/nose/issues/24

**Comment by jchodera** [comment]
> Wait, that is not a real fix though, because it disables the yank setup command. Still need a complete workaround.

Sorry, brain was not working. You are totally right.

**Comment by rmcgibbo** [comment]
Rename setup to something else?

**Comment by kyleabeauchamp** [comment]
I pinged the nose github issue to see if people have a good workaround.

Worst-case workaround is perhaps rename `yank setup` to `yank prepare`

**Comment by jchodera** [comment]
OK, sounds like a good fix for now.

**Comment by kyleabeauchamp** [comment]
This was fixed in #189.
